Controllers/user.js

```diff
@@ -224,6 +224,17 @@ async function toggleStatus(req,res){
   }
 }
 
+async function addLocation(req,res){
+  var query = {_id: new ObjectId(req.params.id)};
+  try{
+    var result = await UserModel.findOneAndUpdate(query, {location: {lat: req.body.lat, lon: req.body.lon}}, {useFindAndModify: false});
```

Comment:
What does useFindAndModify do here?

Reply:
findAndModify is a mongoDB method, and mongoose gives a deprecation warning if I don't include {useFindAndModify: false} as an option

Reply:
Can you set it globally with
`mongoose.set('useFindAndModify', false);`

Reply:
I don't think so; {useFindAndModify: false} is an object used as a parameter (labeled as options in the Mongoose API, you can look it up) for the findOneAndUpdate function (which is a method for Models in Mongoose, you can look it up). Feel free to look up the Mongoose API docs for clarifications

Reply:
https://mongoosejs.com/docs/deprecations.html#findandmodify

Reply:
ahhhhhhhhh I see, thanks

Reply:
check the new commits, fixed it

Reply:
check the new commits, fixed it